File: src/net/reichholf/dreamdroid/helpers/enigma2/Message.java
/* © 2010 Stephan Reichholf <stephan at reichholf dot net>
 * 
 * Licensed under the Create-Commons Attribution-Noncommercial-Share Alike 3.0 Unported
 * http://creativecommons.org/licenses/by-nc-sa/3.0/
 */

package net.reichholf.dreamdroid.helpers.enigma2;

import java.util.ArrayList;

import net.reichholf.dreamdroid.helpers.ExtendedHashMap;

import org.apache.http.NameValuePair;
import org.apache.http.message.BasicNameValuePair;

/**
 * @author sreichholf
 *
 */
public class Message extends SimpleResult {
	public static final String TEXT = "message";
	public static final String TYPE = "type";
	public static final String TIMEOUT = "timeout";	
	public static final String MESSAGE_TYPE_WARNING = "1";
	public static final String MESSAGE_TYPE_INFO = "2";
	public static final String MESSAGE_TYPE_ERROR = "3";
	
//	public static String send(SimpleHttpClient shc, ExtendedHashMap message) {
//		
//		ArrayList<NameValuePair> params = new ArrayList<NameValuePair>();
//		params.add(new BasicNameValuePair("text", message.getString(TEXT)));
//		params.add(new BasicNameValuePair("type", message.getString(TYPE)));
//		params.add(new BasicNameValuePair("timeout", message.getString(TIMEOUT)));
//		
//		if (shc.fetchPageContent(URIStore.MESSAGE, params)) {
//			return shc.getPageContentString();
//		}
//
//		return null;
//	}
	
	public static ArrayList<NameValuePair> getParams(ExtendedHashMap message){
		ArrayList<NameValuePair> params = new ArrayList<NameValuePair>();
		params.add(new BasicNameValuePair("text", message.getString(TEXT)));
		params.add(new BasicNameValuePair("type", message.getString(TYPE)));
		params.add(new BasicNameValuePair("timeout", message.getString(TIMEOUT)));
		
		return params;
	}
}

File: app/src/net/reichholf/dreamdroid/helpers/enigma2/Message.kt
package net.reichholf.dreamdroid.helpers.enigma2

import net.reichholf.dreamdroid.helpers.NameValuePair

object Message {
    const val KEY_TEXT: String = "message"
    const val KEY_TYPE: String = "type"
    const val KEY_TIMEOUT: String = "timeout"
    const val MESSAGE_TYPE_WARNING: String = "1"
    const val MESSAGE_TYPE_INFO: String = "2"
    const val MESSAGE_TYPE_ERROR: String = "3"

    fun getParams(text: String?, type: String?, timeout: String?): ArrayList<NameValuePair> {
        val params = ArrayList<NameValuePair>()
        params.add(NameValuePair("text", text))
        params.add(NameValuePair("type", type))
        params.add(NameValuePair("timeout", timeout))
        return params
    }
}
